src/main/java/org/opentripplanner/service/realtimevehicles/internal/DefaultRealtimeVehicleService.java

```diff
@@ -0,0 +1,64 @@
+package org.opentripplanner.service.realtimevehicles.internal;
+
+import static org.opentripplanner.transit.model.timetable.OccupancyStatus.NO_DATA_AVAILABLE;
+
+import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
+import java.time.Instant;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.annotation.Nonnull;
+import org.opentripplanner.service.realtimevehicles.RealtimeVehicleRepository;
+import org.opentripplanner.service.realtimevehicles.RealtimeVehicleService;
+import org.opentripplanner.service.realtimevehicles.model.RealtimeVehicle;
+import org.opentripplanner.transit.model.network.TripPattern;
+import org.opentripplanner.transit.model.timetable.OccupancyStatus;
+import org.opentripplanner.transit.model.timetable.Trip;
+import org.opentripplanner.transit.service.TransitService;
+
+@Singleton
+public class DefaultRealtimeVehicleService
+  implements RealtimeVehicleService, RealtimeVehicleRepository {
+
+  private final Map<TripPattern, List<RealtimeVehicle>> vehicles = new ConcurrentHashMap<>();
+
+  private final TransitService transitService;
+
+  @Inject
+  public DefaultRealtimeVehicleService(TransitService transitService) {
+    this.transitService = transitService;
+  }
```

Comment:
At todays meeting, it was mentioned that we could inject a function here instead - I do not like that. This is way:

 - Dependency injection get a bit complicated and the relationship between module are not clear. The RealtimeVehicleService could define a role, and then we would wire in that role - but that is in this case overkill. 
 
So, the question then is - How do we make this unit testable. The solution is to refactor and separate the wiring from the logic. If we are pragmatic about it we would allow the test to provide the service we inject by overriding a method (`getPatternForTrip`) or simpler just do this:
```
@Nonnull
  @Override
  public OccupancyStatus getVehicleOccupancyStatus(@Nonnull Trip trip) {
    return findOccupancyStatus(trip.getId(), transitService.getPatternForTrip(trip));
  }

  OccupancyStatus findOccupancyStatus(FeedScopedId tripId, TripPattern pattern) {
    return vehicles
      .getOrDefault(pattern, List.of())
      .stream()
      .filter(vehicle -> tripId.equals(vehicle.trip().getId()))
      .max(Comparator.comparing(vehicle -> vehicle.time().orElse(Instant.MIN)))
      .map(RealtimeVehicle::occupancyStatus)
      .orElse(Optional.empty())
      .orElse(NO_DATA_AVAILABLE);
  }
  ```
I would then test the second method only, the first does not contain a singe branch only wiring , so writing test on it is no point - what do you test => you test ;-)

Reply:
When refactoring this I noticed that we find the `occupancyStatus` for a Trip, but we do not pass in `serviceDate`. Is it ok? It probably at least deserves a comment - does the service only contain information for the past 24hours ?

Reply:
Another thing I would like to test which currently has a transit service dependency is the GtfsRealtimeFuzzyTripMatcher. There are already tests for the class itself but I would like to also have tests that the fuzzy trip matching works in vehicle position parsing.

The service contains information about all the vehicles which are included in a feed but I think the feeds should drop the vehicles after a trip is over. We don't store historic data in the service. The javadoc on the service interface method currently says "Get the latest occupancy status for a certain trip." but I could make it more clear. In the schema, I have written "The latest realtime occupancy information for the latest occurance of this trip." which should be true but now when I think about it, the vehicle could show up on the feed slightly before the trip starts and then this could be interpreted to refer to a trip from a previous day.